flowmachine/flowmachine/features/subscriber/meaningful_locations.py

```diff
@@ -0,0 +1,259 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from typing import Dict, Any, List, Union
+
+from flowmachine.core.errors import BadLevelError
+from ...core import GeoTable, Query
+from . import LabelEventScore, HartiganCluster, EventScore
+from ..spatial import Grid
+from ...utils.utils import get_columns_for_level
+
+
+class MeaningfulLocations(Query):
+    """
+    Infer 'meaningful' locations for individual subscribers (for example, home and work) based on
+    a clustering of the cell towers they use, and their usage patterns for those towers.
+
+    This is an extension of work by Isaacman et al.[1]_ by Flowminder in collaboration with the World Bank[2]_.
+
+    Parameters
+    ----------
+    clusters : HartiganCluster
+        Per subscriber clusters of towers
+    scores : EventScore
+        Per user, per tower scores based on hour of day and day of week of interactions with the tower
+    labels : LabelEventScore
```

Comment:
In this class the difference between the `labels` and `label` argument is a little clearer, but probably still worth expanding on it (or giving an example in the docstring).